crates/store/src/state.rs

```diff
@@ -96,17 +96,107 @@ pub struct TransactionInputs {
     pub found_unauthenticated_notes: BTreeSet<NoteId>,
 }
 
+/// Blockchain wrapper for `Mmr`
+#[derive(Debug, Clone)]
+pub struct Blockchain(Mmr);
+
+impl Blockchain {
+    /// Returns the latest block number
+    pub fn chain_tip(&self) -> BlockNumber {
+        let block_number: u32 = (self.0.forest() - 1)
+            .try_into()
+            .expect("chain_mmr always has, at least, the genesis block");
+
+        block_number.into()
+    }
+
+    /// Returns the chain length
+    pub fn chain_length(&self) -> BlockNumber {
+        let latest = self.chain_tip();
+        latest + 1
+    }
+
+    /// Returns the current peaks of the MMR.
+    pub fn peaks(&self) -> MmrPeaks {
+        self.0.peaks_at(self.0.forest()).expect("failed to get peaks at current forest")
+    }
+
+    /// Returns the peaks of the MMR at the state specified by `forest`.
+    ///
+    /// # Errors
+    /// Returns an error if the specified `forest` value is not valid for this MMR.
+    pub fn peaks_at(&self, forest: usize) -> Result<MmrPeaks, MmrError> {
+        self.0.peaks_at(forest)
+    }
+
+    /// Adds a new element to the MMR.
+    pub fn add(&mut self, el: RpoDigest) {
+        self.0.add(el);
+    }
+
+    /// Returns an [`MmrProof`] for the leaf at the specified position
+    pub fn open(&self, pos: usize) -> Result<MmrProof, MmrError> {
+        self.0.open_at(pos, self.0.forest())
+    }
+
+    /// Returns the latest block number and partial mmr
+    pub fn chain_mmr(
```

Comment:
This actually only constructs a partial MMR now, so we should name it accordingly, e.g. `partial_mmr_from_blocks` or something.